Prevent damage to Extendo-Grip from Frost Walker

cassiancc opened the pull request:
Frost Walker places frosted ice blocks through the EntityPlaceEvent. Currently, the event then damages the Extendo-Grip for every frosted ice block placed by the enchantment, rapidly breaking it. This fix prevents damage to the Extendo-Grip when the block is Frosted Ice and the player's equipment has Frost Walker.

cassiancc committed:
Prevent damage to Extendo-Grip from Frost Walker

Frost Walker places frosted ice blocks through the EntityPlaceEvent. Currently, the event then damages the Extendo-Grip for every frosted ice block placed by the enchantment, rapidly breaking it. This fix prevents damage to the Extendo-Grip when the block is Frosted Ice and the player's equipment has Frost Walker.

cassiancc commented:
This should fix #5494.

VoidLeech reviewed (commented):
You should have the call to findAndDamageExtendoGrip not be in an else block, and just do an early return. As is, this is going to call findAndDamageExtendoGrip if any armor piece has any enchantment that isn’t Frost Walker, and multiple times as well. 
Also, there should just be a method in EnchantmentHelper that’ll tell you whether an entity has an enchantment, no need to iterate over armorslots and enchantments. (And if iterating over armorslots was necessary, there’s always Map#containsKey)

cassiancc committed:
Alternate fix

Checking to see if the block is Frosted Ice is failing, possibly due to the event happening before the block is placed.

TropheusJ commented:
this will deal no damage when the player is placing blocks in water.

ChiefArug commented:
Frosted Ice does not have an item form (in vanilla) so is unplacable by players from their hand by default (which is what the extendo grip is trying to detect here, blocks being placed by the player's hand), surely its enough to just check if the placed block is frosted ice?
There is also a decent chance that other mods have a similar effect implemented in the same way (ie frost walker for lava), so it would be worth using a block tag for this check instead of a hardcoded `is` check so that this behaviour can be customized.
